File: javascript/assignment2.js
let distance = 550; // You can change this number to test different tiers
let amountToPay;

if (distance >= 0 && distance <= 100) {
    amountToPay = 5;
} else if (distance >= 101 && distance <= 500) {
    amountToPay = 10;
} else if (distance >= 501 && distance <= 1000) {
    amountToPay = 20;
} else if (distance >= 1001) {
    amountToPay = 40;
} else {
    console.log("Invalid distance entered.");
}

console.log("Your total is: " ,amountToPay ," USD");

//triple equal sign-research
//terms, === (Triple Equal) is a "strict" check. It only says True if the value and the type (Number vs. String) are exactly the same.

let price = 100;      // This is a Number
let input = "100";    // This is a String (text)

console.log(price == input);  // Returns: True  (Lazy check)
console.log(price === input); // Returns: False (Strict check - Types don't match)

//  what is the difference between a dictionary in python,an object in javascript and a json data when dealing with APIs

//A Python Dictionary and JavaScript Object are "live" containers used inside their respective languages to hold and change data while the program is running. JSON is not a container, but a text-only format (a string) used to ship that data across the internet between the two.

File: kotlin/Assignment2.kt
// Program 2: Check if a number is Odd or Even

fun main() {
    val number = 15  // Change this value to test different numbers
    
    if (number % 2 == 0) {
        println("$number is EVEN")
    } else {
        println("$number is ODD")
    }
    
    // Test multiple numbers
    println("\nTesting multiple numbers:")
    val numbers = arrayOf(10, 15, 22, 7, 0, -5, 100, 99)
    for (num in numbers) {
        if (num % 2 == 0) {
            println("$num is EVEN")
        } else {
            println("$num is ODD")
        }
    }
}

// Program 1: Leap Year Checker using if statements

fun main() {
    val year = 2024  // Change this value to test different years
    
    if (year % 4 == 0) {
        if (year % 100 == 0) {
            if (year % 400 == 0) {
                println("$year is a LEAP year")
            } else {
                println("$year is NOT a leap year")
            }
        } else {
            println("$year is a LEAP year")
        }
    } else {
        println("$year is NOT a leap year")
    }
    
    // Test multiple years
    println("\nTesting multiple years:")
    val years = arrayOf(2000, 2020, 2024, 1900, 2023, 2100)
    for (y in years) {
        val result = if (y % 4 == 0) {
            if (y % 100 == 0) {
                if (y % 400 == 0) "LEAP year" else "NOT a leap year"
            } else {
                "LEAP year"
            }
        } else {
            "NOT a leap year"
        }
        println("$y is $result")
    }
}